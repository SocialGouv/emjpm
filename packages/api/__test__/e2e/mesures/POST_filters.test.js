//

const request = require("supertest");

const { knex } = global;
jest.setMock("@emjpm/api/db/knex", knex);

const server = require("@emjpm/api/app");

const { getTokenByUserType } = require("../utils");

//

beforeAll(async () => {
  await knex.migrate.latest();
  await knex.seed.run();
});

// strip the created_at date in the response
const simpler = ({ created_at, ...props }) => props;

test("should get mesures filtered by localisation", async () => {
  const token = await getTokenByUserType("ti");
  const response = await request(server)
    .post("/api/v1/mesures/filters")
    .set("Authorization", "Bearer " + token)
    .send({
      latNorthEast: 1,
      latSouthWest: 0,
      longNorthEast: 1,
      longSouthWest: 0
    });

  expect(response.body.map(simpler)).toMatchSnapshot();

  expect(response.status).toBe(200);
});
