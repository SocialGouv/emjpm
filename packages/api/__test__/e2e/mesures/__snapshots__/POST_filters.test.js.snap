// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get mesures filtered by localisation 1`] = `
Array [
  Object {
    "adresse": "21 rue de houx",
    "cabinet": null,
    "code_postal": "62000",
    "contact_email": null,
    "contact_nom": null,
    "contact_prenom": null,
    "date_mesure_update": null,
    "dispo_max": 5,
    "email_send": null,
    "etablissement": "UDAHF",
    "genre": null,
    "id": 1,
    "latitude": 0.3,
    "longitude": 0.3,
    "mesures_en_attente": null,
    "mesures_en_cours": 2,
    "nb_secretariat": 2,
    "nom": null,
    "prenom": "Adrien",
    "secretariat": false,
    "service_dispo_max": null,
    "service_email": null,
    "service_etablissement": null,
    "service_id": null,
    "service_nom": null,
    "service_prenom": null,
    "service_telephone": null,
    "telephone": "0237100000",
    "telephone_portable": "0101010101",
    "type": "individuel",
    "user_id": 1,
    "ville": "Arras",
    "zip": null,
  },
  Object {
    "adresse": "21 rue monge",
    "cabinet": null,
    "code_postal": "62000",
    "contact_email": null,
    "contact_nom": null,
    "contact_prenom": null,
    "date_mesure_update": null,
    "dispo_max": 3,
    "email_send": null,
    "etablissement": "test42",
    "genre": null,
    "id": 2,
    "latitude": 0.2,
    "longitude": 0.2,
    "mesures_en_attente": null,
    "mesures_en_cours": 2,
    "nb_secretariat": 2,
    "nom": null,
    "prenom": "Julien",
    "secretariat": true,
    "service_dispo_max": null,
    "service_email": null,
    "service_etablissement": null,
    "service_id": null,
    "service_nom": null,
    "service_prenom": null,
    "service_telephone": null,
    "telephone": "0237100000",
    "telephone_portable": "0101010101",
    "type": "prepose",
    "user_id": 2,
    "ville": "Paris",
    "zip": null,
  },
]
`;
